components/apimgt/org.wso2.carbon.apimgt.keymgt/src/main/java/org/wso2/carbon/apimgt/keymgt/token/AbstractJWTGenerator.java

```diff
@@ -216,7 +218,18 @@ public String buildBody(TokenValidationContext validationContext) throws APIMana
                     String claimURI = it.next();
                     String claimVal = standardClaims.get(claimURI);
                     List<String> claimList = new ArrayList<String>();
-                    if (userAttributeSeparator != null && claimVal != null && claimVal.contains(userAttributeSeparator)) {
+                    if (claimVal != null && claimVal.contains("{")) {
+                        ObjectMapper mapper = new ObjectMapper();
+                        try {
+                            Map<String, String> map = mapper.readValue(claimVal, Map.class);
+                            jwtClaimsSetBuilder.claim(claimURI, map);
+                        } catch (IOException e) {
+                            // Exception isn't thrown in order to generate jwt without claim, even if an error is
+                            // occurred during the retrieving claims.
+                            log.error("Error while reading claim values", e);
```

Comment:
We should add some more details to this error message, what claim value, URI failed, what user etc if possible.

Reply:
Fixed with 4158a39f601c8feed0db799983f4f5bdf49a1f5d